src/main.rs

```diff
@@ -264,14 +263,32 @@ fn handle_dai_balance_update(
     unimplemented!()
 }
 
+// TODO: I don't think `finished_swap` should be an Option
```

Comment:
What would `None` mean?
From the reserve point of view, we have to unreserve (and "unlock" the taker from ongoing swaps) in all these three cases (because it was reserved either way and has to be freed):

- Taker never funded (i.e. swap timed out)
- Refunded
- Redeemed

Since we have to react to all these scenarios in the same way, I wonder what `None` would represent.

Reply:
AFAIK `None` means that the stream is terminated.